snuba/datasets/plans/storage_processing.py

```diff
@@ -162,7 +163,7 @@ def apply_storage_processors(
                 storage_key=storage.get_storage_key(),
             )
         )
-
+    assert isinstance(query_plan.query, Query)
```

Comment:
A query plan object can either contain a query that is of type `ClickHouseQuery` or `ProcessableQuery`. We need to support both because of the simple and composite query pipelines respectively. This function `apply_storage_processors` accepts a query plan object as an argument, and is only used in the simple query pipeline. As a result, we need to type check the query in order to run `processor.process_query`.